src/lang-promql/complete/hybrid.ts

```diff
@@ -134,17 +132,33 @@ export class HybridComplete implements CompleteStrategy {
       // So we can autocomplete the labelValue
       return this.autocompleteLabelValue(tree.parent, tree, pos, context, state);
     }
+    if (tree.type.id === LabelMatcher && tree.firstChild?.type.id === LabelName && tree.lastChild !== null) {
+      // In this case the current token is not itself a valid match op yet:
+      //      metric_name{labelName!}
+      return this.arrayToCompletionResult([autocompleteNode.matchOp], tree.lastChild.start, pos, context, state);
+    }
     if (tree.name === 'MatchOp') {
-      return this.arrayToCompletionResult([autocompleteNode[matchOp]], tree.start, pos, context, state);
+      // In this case the current token is already a valid match op, but could be extended, e.g. "=" to "=~".
+      return this.arrayToCompletionResult([autocompleteNode.matchOp], tree.start, pos, context, state);
     }
     if (tree.parent?.type.id === BinaryExpr) {
-      return this.arrayToCompletionResult([autocompleteNode[BinaryExpr]], tree.start, pos, context, state);
+      return this.arrayToCompletionResult([autocompleteNode.binaryExpr], tree.start, pos, context, state);
     }
     if (tree.parent?.type.id === FunctionIdentifier) {
-      return this.arrayToCompletionResult([autocompleteNode[FunctionIdentifier]], tree.start, pos, context, state);
+      return this.arrayToCompletionResult([autocompleteNode.functionIdentifier], tree.start, pos, context, state);
     }
     if (tree.parent?.type.id === AggregateOp) {
-      return this.arrayToCompletionResult([autocompleteNode[AggregateOp]], tree.start, pos, context, state);
+      return this.arrayToCompletionResult([autocompleteNode.aggregateOp], tree.start, pos, context, state);
+    }
+    if ((tree.type.id === Identifier && tree.parent?.type.id === 0) || (tree.type.id === 0 && tree.parent?.type.id !== LabelMatchers)) {
+      // This matches identifier-ish keywords in certain places where a normal identifier would be invalid, like completing "b" into "by":
+      //        sum b
+      // ...or:
+      //        sum(metric_name) b
+      // ...or completing "unle" into "unless":
+      //        metric_name / unle
+      // TODO: This is imprecise and autocompletes in too many situations. Make this better.
+      return this.arrayToCompletionResult([autocompleteNode.aggregateOpModifier].concat(autocompleteNode.binOp), tree.start, pos, context, state);
```

Comment:
here does it make sense to autocomplete the metrics name too ?

Reply:
No, these are places where a normal expression (of which a metric name or function call are examples) would be invalid. E.g. you can't have `metric_name1 metric_name2`, that's not a valid PromQL expression.